File: pset1/credit.c
#include<cs50.h>
#include<stdio.h>

int main (void){
    long long i;
    do
    {
        i = get_long_long("Number: ");

    }
    while(i<0);
    int sum, first, second;
    int count = 1;
    while(i!=0){
        int d = i%10;
        i =i/10;

        if(count%2==0){
            first = d;
            d= d*2;
            if(d>9){d=d/10+d%10;}
            sum = sum+d;
            count++;
        }
        else{
        sum = sum+d;
        count++;
        second = d;
        }
    }
    count = count-1;

    if(!(count%2==0)){
        int n = first;
        first = second;
        second =n;
    }

    if(sum%10==0){
        if((count ==15 && first == 3) && (second==4||second==7)){
            printf("AMEX\n");
        }

        else if((count ==16 || count==13) && first==4){
            printf("VISA\n");
        }

        else if((first==5 && count==16) && (second<6 && second!=0)){
            printf("MASTERCARD\n");
         }

         else{
             printf("INVALID\n");
        }


    }
    else{
        printf("INVALID\n");

    }

}

File: pset6/sentimental/credit.py
from cs50 import get_int

def main():
    print("Number: ", end="")
    while True:
        i = get_int()
        if i > 0:
            break

    count = 1
    sum = 0
    while i != 0:
        d = i%10
        i=i//10

        if count%2 == 0:
            first = d
            d = d*2
            if d>9:
                d=d//10+d%10
            sum += d
            count+=1
        else:
            sum +=d
            count+=1
            second = d
    count-=1
    if count%2 != 0:
        first, second = second, first

    if sum%10 == 0:
        if (count == 15 and first == 3) and (second == 4 or second ==7):
            print("AMEX")

        elif ((count ==16 or count==13) and first==4):
            print("VISA")

        elif ((first==5 and count==16) and (second<6 and second!=0)):
            print("MASTERCARD")

        else:
            print("INVALID")
    else:
        print("INVALID")

if __name__ == "__main__":
    main()